lgc/include/lgc/state/PipelineState.h

```diff
@@ -552,7 +552,7 @@ class PipelineState final : public Pipeline {
   bool m_preRasterHasGs = false;                        // Whether pre-rasterization part has a geometry shader
   bool m_computeLibrary = false;                        // Whether pipeline is in fact a compute library
   std::string m_client;                                 // Client name for PAL metadata
-  Options m_options = {};                               // Per-pipeline options
```

Comment:
`Options m_options = Options()` should also work, because this value-initializes instead of aggregate initialization [1]. In my local test, this worked with both gcc and clang, while `= {}` did not initialize padding with clang.

However, even if this works now, I feel the memset variant is more robust.

[1]: https://stackoverflow.com/questions/70979077/does-c-standard-guarantee-the-initialization-of-padding-bytes-to-zero-for-non

Reply:
But the stackoverflow link says the `Foo()` does not work for g++, right?

Reply:
Seems to be a bug, so yeah, let's not depend on this.